### PR TITLE
sql: validate UseDeletePreservingEncoding is never public

### PR DESCRIPTION
**Pull request by stevendanna**
This validates that an index with `UseDeletePreservingEncoding` set to
true is never made public.

To facilitate this, I moved some of the tests that use this encoding
from execbuilder into the same test we added for the merging index
tests. The only changes to the kvtraces were

- TableID changes

- Since the index isn't public, it isn't scanned in one case where
  previously it was scanned.

Release note: None

### CODE REVIEW

File: pkg/sql/force_put_index_test.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"context"
```
Comment (mgartner) [inline pkg/sql/force_put_index_test.go:14]: Should we rename this file now that it tests more than force PUTs?

**Review by mgartner — APPROVED** [review]
LGTM

**Review by ajwerner — APPROVED** [review]
:lgtm: mod CI

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/75945#-:-Mv-RCGb0H52srnH7giW:b-yef5vx)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rhu713 and @stevendanna)

**Comment by stevendanna** [comment]
bors r=ajwerner

**Comment by stevendanna** [comment]
bors failure: https://github.com/cockroachdb/cockroach/pull/76125

**Comment by stevendanna** [comment]
bors cancel

**Comment by stevendanna** [comment]
bors r=ajwerner

**Comment by stevendanna** [comment]
bors retry

**Pull request merged**